org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/JavaDocCompletionProposal.java

```diff
@@ -0,0 +1,264 @@
+import org.eclipse.jdt.ls.core.internal.corext.codemanipulation.StubUtility;
+import org.eclipse.jdt.ls.core.internal.handlers.CompletionResolveHandler;
+import org.eclipse.jdt.ls.core.internal.handlers.JsonRpcHelpers;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.Region;
+import org.eclipse.jface.text.TextUtilities;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionItemKind;
+import org.eclipse.lsp4j.InsertTextFormat;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+public class JavaDocCompletionProposal {
+
+	public static final String JAVA_DOC_COMMENT = "JavaDoc comment";
+
+	public List<CompletionItem> getProposals(ICompilationUnit cu, int offset, CompletionProposalRequestor collector, IProgressMonitor monitor) throws JavaModelException {
+		if (cu == null) {
+			throw new IllegalArgumentException("Compilation unit must not be null"); //$NON-NLS-1$
+		}
+		List<CompletionItem> result = new ArrayList<>();
+		IDocument d = JsonRpcHelpers.toDocument(cu.getBuffer());
+		if (offset < 0 || d.getLength() == 0) {
+			return result;
+		}
+		try {
+			int p = (offset == d.getLength() ? offset - 1 : offset);
+			IRegion line = d.getLineInformationOfOffset(p);
+			String lineStr = d.get(line.getOffset(), line.getLength()).trim();
+			if (!lineStr.startsWith("/**")) {
+				return result;
+			}
+			if (!hasEndJavadoc(d, offset)) {
+				return result;
+			}
+			String text = collector.getContext().getToken() == null ? "" : new String(collector.getContext().getToken());
+			StringBuilder buf = new StringBuilder(text);
+			IRegion prefix = findPrefixRange(d, line);
+			String indentation = d.get(prefix.getOffset(), prefix.getLength());
+			int lengthToAdd = Math.min(offset - prefix.getOffset(), prefix.getLength());
+			buf.append(indentation.substring(0, lengthToAdd));
+			String lineDelimiter = TextUtilities.getDefaultLineDelimiter(d);
+			ICompilationUnit unit = cu;
+			try {
+				unit.reconcile(ICompilationUnit.NO_AST, false, null, null);
+				String string = createJavaDocTags(d, offset, indentation, lineDelimiter, unit);
+				if (string != null && !string.trim().equals("*")) {
+					buf.append(string);
+				} else {
+					return result;
+				}
+				int nextNonWS = findEndOfWhiteSpace(d, offset, d.getLength());
+				if (!Character.isWhitespace(d.getChar(nextNonWS))) {
+					buf.append(lineDelimiter);
+				}
+			} catch (CoreException e) {
+				// ignore
+			}
+			final CompletionItem ci = new CompletionItem();
+			Range range = JDTUtils.toRange(unit, offset, 0);
+			ci.setTextEdit(new TextEdit(range, buf.toString()));
+			ci.setFilterText(JAVA_DOC_COMMENT);
+			ci.setLabel(JAVA_DOC_COMMENT);
+			ci.setSortText(SortTextHelper.convertRelevance(0));
+			ci.setKind(CompletionItemKind.Snippet);
+			ci.setInsertTextFormat(InsertTextFormat.Snippet);
```

Comment:
CompletionItemKind.Snippet is fine, but I don't think InsertTextFormat.Snippet is appropriate here, since there will be no placeholders. Clients not supporting Snippets might puke when receiving those items. We should send  InsertTextFormat.PlainText